pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -1236,8 +1236,28 @@ type AzureAD struct {
 	// +optional
 	Cloud *string `json:"cloud,omitempty"`
 	// ManagedIdentity defines the Azure User-assigned Managed identity.
-	// +required
+	// Cannot be set at the same time as oauth.
```

Comment:
```suggestion
	// Cannot be set at the same time as `oauth`.
```

For consistency with other areas we have similar behavior :)